src/common/java/net/minecraftforge/gradle/common/task/DownloadAssets.java

```diff
@@ -53,13 +54,15 @@ public void run() throws IOException, InterruptedException {
         for (String key : keys) {
             Asset asset = index.objects.get(key);
             File target = Utils.getCache(getProject(), "assets", "objects", asset.getPath());
-            if (!target.exists()) {
+            if (!target.exists() || !HashFunction.SHA1.hash(target).equals(asset.hash)) {
                 URL url = new URL(RESOURCE_REPO + asset.getPath());
                 Runnable copyURLtoFile = () -> {
                     try {
                         getProject().getLogger().lifecycle("Downloading: " + url + " Asset: " + key);
                         FileUtils.copyURLToFile(url, target, 10_000, 5_000);
-
+                        if (!HashFunction.SHA1.hash(target).equals(asset.hash)) {
+                            throw new IOException(key + " Hash dose march");
```

Comment:
Typo here, should probably just be "Hash failed."

Reply:
OK.